client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -17,151 +17,208 @@ import {
 	isRequestingSitePlans,
 } from 'state/sites/plans/selectors';
 import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
 import Button from 'components/button';
 import Card from 'components/card';
 import MyPlanCard from './my-plan-card';
 import QuerySites from 'components/data/query-sites';
 import QuerySitePlans from 'components/data/query-site-plans';
 import QuerySitePurchases from 'components/data/query-site-purchases';
+import { managePurchase } from 'me/purchases/paths';
 import { getPlan } from 'lib/plans';
-import { getByPurchaseId } from 'state/purchases/selectors';
 import { isPartnerPurchase, shouldAddPaymentSourceInsteadOfRenewingNow } from 'lib/purchases';
-import { isFreeJetpackPlan, isFreePlan } from 'lib/products-values';
-import { managePurchase } from 'me/purchases/paths';
+import {
+	isFreeJetpackPlan,
+	isFreePlan,
+	isJetpackBackup,
+	isProductExpiring,
+	getJetpackProductDisplayName,
+	getJetpackProductTagLine,
+} from 'lib/products-values';
 
 class PurchasesListing extends Component {
 	static propTypes = {
 		currentPlan: PropTypes.object,
 		currentPlanSlug: PropTypes.string,
 		isPlanExpiring: PropTypes.bool,
-		isRequestingSitePlans: PropTypes.bool,
+		isRequestingPlans: PropTypes.bool,
 		selectedSite: PropTypes.object,
 		selectedSiteId: PropTypes.number,
 		selectedSiteSlug: PropTypes.string,
-		purchase: PropTypes.object,
+		purchases: PropTypes.array,
 
 		// From localize() HoC
 		translate: PropTypes.func.isRequired,
 	};
 
 	isLoading() {
-		const { currentPlanSlug, selectedSite, isRequestingSitePlans: isRequestingPlans } = this.props;
+		const { currentPlan, selectedSite, isRequestingPlans } = this.props;
 
-		return ! currentPlanSlug || ! selectedSite || isRequestingPlans;
+		return ! currentPlan || ! selectedSite || isRequestingPlans;
 	}
 
-	isFreePlan() {
+	isFreePlan( purchase ) {
 		const { currentPlan } = this.props;
 
+		if ( purchase && isJetpackBackup( purchase ) ) {
+			return false;
+		}
+
 		return ! currentPlan || isFreePlan( currentPlan ) || isFreeJetpackPlan( currentPlan );
 	}
 
-	getPlanObject() {
+	getTitle( purchase ) {
 		const { currentPlanSlug } = this.props;
 
-		if ( ! currentPlanSlug ) {
+		if ( isJetpackBackup( purchase ) ) {
+			return getJetpackProductDisplayName( purchase );
+		}
+
+		if ( currentPlanSlug ) {
+			const planObject = getPlan( currentPlanSlug );
+			return planObject?.getTitle?.();
```

Comment:
I discourage against using optional chaining for cases like this. If there's no plan object here, then there's something else that's wrong. And if the plan object doesn't have `getTitle` there's something else wrong. We shouldn't be concealing these errors, instead we should be making sure they don't occur (each plan object MUST have a `getTitle()` method).

Reply:
:+1:

Reply:
Makes sense to me 👍